src/IO/S3Common.cpp

```diff
@@ -90,8 +91,23 @@ namespace S3
         const String & secret_access_key)
     {
         Aws::Client::ClientConfiguration cfg;
+
         if (!endpoint.empty())
+        {
+            static const RE2 region_pattern(R"(s3\.([a-z0-9\-]+)\.amazonaws\.)");
```

Comment:
Hard coded amazonaws seems not viable. There are many S3 compatible object storage providers

Reply:
I guess even for S3 it is not complete... How would you recommend to determine the region?

Reply:
It is used only for region detection. For now we don't know anything about requirements to pass/support region for non aws s3. 


region
Specifies the AWS region to use, such as us-east-1. By default, the region used is the default region configured in the applicable AWS credentials.

Reply:
I.e. If regexp will not match we just not pass the region, and everything works as before for non-aws

Reply:
Ok.